### PR TITLE
etcd: fix JWT renewal with user/pass authentication

### LINKED ISSUE #6881: etcd: infinite reconnect loop when using JWT authentication
Our etcd client library is missing https://github.com/etcd-io/etcd/pull/12165

This causes teleport to go into an infinite reconnect loop within its etcd client once the initial JWT expires.
This situation only happens when:
- teleport uses usrname/password authentication for etcd
- etcd cluster is configured with `--auth-token=jwt` (https://etcd.io/docs/v3.3/op-guide/configuration/#--auth-token)

Update the etcd client library if possible.
The fallback would be some hacky grpc interceptor hook into the etcd client to detect this and recreate the entire client.

### PR DESCRIPTION
**Pull request by awly**
Two changes here:

1. update go.etcd.io/etcd to their latest module-compliant versions
2. use a separate connection for peer version checking
    the implementation of `b.client.Status` (in the etcd library) internally causes the in-memory JWT store to be overwritten.
    this ends up breaking the JWT refresh logic (one store has the exired token but a different one gets the refreshed one).
    I will send an upstream fix too

\# 2 works without # 1.
Since this PR is against `master`, I decided to keep # 1 anyway, for 7.0.
I will cherry-pick only # 2 to 6.2, without dependency changes.

FYI @fspmarshall because you had some memory issues with the etcd client; check if this updated version of the library fixes them

Fixes https://github.com/gravitational/teleport/issues/6881

### CODE REVIEW

**Comment by Joerger** [comment]
@klizhentas has shown concern about moving to a beta version of etcd for a production release #6183. However it doesn't seem clear when etcd plans on releasing 3.5 and this has pushed back a few other fixes as well. @awly Do you have a sense of how stable the current beta version is or how soon it'll be released officially?

File: lib/backend/etcdbk/etcd.go [hunk 1]
```diff
@@ -292,6 +298,12 @@ func (b *EtcdBackend) CloseWatchers() {
 }
 
 func (b *EtcdBackend) reconnect(ctx context.Context) error {
+	if b.client != nil {
+		if err := b.client.Close(); err != nil {
+			b.Entry.WithError(err).Warningf("Failed closing existing etcd client on reconnect.")
```
Comment (r0mant) [inline lib/backend/etcdbk/etcd.go:303]: Nit: 
```suggestion
			b.Entry.WithError(err).Warning("Failed closing existing etcd client on reconnect.")
```

**Review by r0mant — COMMENTED** [review]
Change lgtm but I agree I would be cautious vendoring an alpha dependency. Is it required for this change?

**Comment by awly** [comment]
You're right, let's wait for etcd 3.5 to be fully released before updating.
Reverted all dependency changes. PTAL

**Review by Joerger — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**